trio/_core/_multierror.py

```diff
@@ -201,7 +257,9 @@ def __init__(self, exceptions, *, _collapse=True):
 
         super().__init__("multiple tasks failed", exceptions)
 
-    def __new__(cls, exceptions, *, _collapse=True):
+    def __new__(
+        cls, exceptions: Iterable[BaseException], *, _collapse: bool = True
+    ) -> MultiError | Self:
```

Comment:
This should probably have an `@overload` using `Literal[False]` - if collapse=False, it will return `MultiError`. Otherwise, it may return any exception.

Reply:
Kind of did in [b21b293](https://github.com/python-trio/trio/pull/2742/commits/b21b2936e03320cffe5454697a67e7ab6e00f799) but not exactly

Reply:
Adding overloads for this ends up failing with type issues from conflicts between overloads. If anyone has any ideas, please share.